bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -317,4 +324,31 @@ public int getRollerFav(int id) {
         }
         return -1;
     }
+
+    public static String extractFwVersion(@Nullable String version) {
+        if (version != null) {
+            Matcher matcher = VERSION_PATTERN.matcher(version);
+            if (matcher.find()) {
+                // e.g. 20210226-091047/v1.10.0-rc2-89-g623b41ec0-master
+                return matcher.group(0);
+            }
+            if (version.contains("@")) { // standard format
+                return substringBetween(getString(version), "/", "@");
+            }
+            if (version.indexOf('/') != version.lastIndexOf('/')) {
+                // includes 2 /, e.g. beta
+                return substringAfterLast(version, "/");
+            }
```

Comment:
You might combine these three variants in one regex. Do you have samples of the other two?

Reply:
ok, I removed if () 2+2. In general the pattern should catch them all, but do to the fact I do not have all variants I left them in as fallback.